daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -2807,17 +2799,171 @@ def forward(node: ONNXOp, state: SDFGState,
                                   memlet=dace.Memlet("sum_res[0]"))
         new_state.add_memlet_path(outer_me, init_tasklet, memlet=dace.Memlet())
 
-
         new_state.add_memlet_path(store_tasklet,
                                   outer_mx,
                                   out_data,
                                   src_conn="out_res",
                                   memlet=dace.Memlet("reduced[o0, o1, o2]"))
 
-
-
-
         new_sdfg.fill_scope_connectors()
         new_sdfg.validate()
         new_sdfg.save('/tmp/reduce_sum.sdfg')
         return new_sdfg
+
+
+# -----------------------------
+# Binary element-wise operators
+# -----------------------------
+def binary_forward(node: ONNXOp, state: SDFGState, sdfg: SDFG,
```

Comment:
Actually, have you seen the Add implementation for CPU/GPU? https://github.com/spcl/daceml/blob/master/daceml/onnx/op_implementations/pure_implementations.py#L108

Couldn't you rewrite it to be similar? I don't know much about the FPGA side of dace, but it seems like you could just use the same implementation + change the map schedule to `FPGA_Device` afterwards.

This would massively reduce code size and give you broadcasting support for free.